### ISSUE TITLE
List of GUI clients

### ISSUE THREAD

**Issue by NightMachinery** [klzgrad/naiveproxy#158]
E.g., https://github.com/KevinZonda/NaiveSharp ,  https://github.com/Qv2ray/Qv2ray . Having a central location to find clients helps a lot.

**Comment by ExcitedCodes** [comment]
Qv2ray might be a good choice for most users, it's really easy to configure and use.
However if you are looking for something that support multiple listeners, NaiveGUI will do better: https://github.com/ExcitedCodes/NaiveGUI/

Typical use case: Use proxifier to handle system traffic and create rules to redirect them to different listeners

**Comment by klzgrad** [comment]
v2rayN claims to support naiveproxy: https://github.com/2dust/v2rayN/commit/c2ea14151e7cbaf0f49da1177d9acf2bd3e7af1e

**Comment by raminxxx** [comment]
Any other app for Android beside of sagernet?

**Comment by klzgrad** [comment]
https://github.com/MatsuriDayo/Matsuri claims support of naiveproxy.

**Comment by raminxxx** [comment]
> https://github.com/MatsuriDayo/Matsuri claims support of naiveproxy.

please look at this :

https://github.com/SagerNet/SagerNet/issues/566

with same setting on sagernet or matsuri naive won't work at all on android for me ( tested with android 13-12-10 )
but on iOS with padding enable it work very very good without padding also in iOS it wont work.

**Comment by U201413497** [comment]
没有一个不需要插件的安卓客户端吗？sagernet和matsuri都还是需要插件才能用，国产的安卓手机在调用插件时候，有的会显示找不到插件，不同的品牌手机设置还不一样，没通用的方法。只有有客户端才好用。

**Comment by U201413497** [comment]
> 

我明白你说的内容，但是Sagernet和matsuri都需要naiveproxy 插件，没有插件是最好的方式。谢谢你的回复。

**Comment by U201413497** [comment]
安卓有开发者在开发无需插件的客户端。https://github.com/Jidoer/AndrProxy

**Comment by bzdk** [comment]
Except Nekoray, is there another GUI client for the macOS?

**Comment by topoge** [comment]
@bzdk 我写了一个桌面客户端，目前支持MacOS，windows，后面会支持Linux
https://github.com/topoge/NgX/releases

**Comment by U201413497** [comment]
https://github.com/ali322/Nave/releases
另外一个支持naiveproxy的客户端

**Comment by ccxxvv77** [comment]
有相关的浏览器插件吗求问

**Comment by U201413497** [comment]
你可以摸索一下switchyomega 插件。

**Comment by ccxxvv77** [comment]
啊 我的意思是直接通过浏览器插件连接Naiveproxy，不是客户端+switchyomega
而SO通过https 设置naive后会提示tunnel错误

**Comment by zedifen** [comment]
@ccxxvv77 If errors is encountered when HTTP proxy is in used, how about giving SOCKS ones a try?

**Comment by zedifen** [comment]
Sorry I misunderstood your claim. So let me put it again, a totally-within-browser proxy extension would be nice for you, is this right?

**Comment by ccxxvv77** [comment]
Yes
And my appreciation if anyone can tell me one

**Comment by klzgrad** [comment]
> 有相关的浏览器插件吗求问

原理上是不可能的，Chrome取消了extension里面访问socket的API。

**Comment by U201413497** [comment]
直接使用Switchyomega,填写https代理,设置账号密码,使用先访问probe_resistance设置的域名就可以使用了。

**Comment by ccxxvv77** [comment]
哦好的谢谢

**Comment by klzgrad** [comment]
> 直接使用Switchyomega,填写https代理,设置账号密码,使用先访问probe_resistance设置的域名就可以使用了。

这不是naiveproxy

**Comment by Chilledheart** [comment]
hi all, i managed to make some naiveproxy-compatible GUI clients both on [iOS][ios] and [Android] (ipa and apk files url). While I tested it on iOS 17 and Android 14, it worked preliminarily as a basic vpn client in the mobile device. Basically, you can test it on iOS 13 or later and Android 7 or later. Please watch out for the full [release note][relnote] before you try it.

Internally, it uses the same SSL stack code from chromium but builds in cmake. I tried to make it as close as chromium but at minimal efforts. It also supports some legacy ciphers from shadowsock but not that much. With the "http2" method, I used it as an alternative to SagerNet in android and Shadowrocket in ios. It is a minimal clone of those GUI clients so it doesn't contain many high level features. Stay tuned and report to me if you want other people to have more support. 

Below is the screenshot of the android client. The screenshot is in Chinese, but don't worry. Of course, it use English as basic language.

<img width="484" alt="snapshot-android" src="https://github.com/Chilledheart/yass/assets/54673341/50d0f9ea-cdcf-409d-830e-9ac846489e14">

The app is quite ugly and basic. It just works as sample as naiveproxy cli command line. For the full family members (it supports windows, linux and macOS), you can look at [screenshot].

[relnote]: https://github.com/Chilledheart/yass/releases/tag/1.5.9
[ios]: https://github.com/Chilledheart/yass/releases/download/1.5.9/yass-ios-release-arm64-1.5.9.ipa
[android]: https://github.com/Chilledheart/yass/releases/download/1.5.9/yass-android-release-arm64-1.5.9.apk
[screenshot]: https://github.com/Chilledheart/yass/wiki/Supported-Operating-System#screenshot-on-android-vpn-support

**Comment by Chilledheart** [comment]
> hi all, i managed to make some naiveproxy-compatible GUI clients both on [iOS](https://github.com/Chilledheart/yass/releases/download/1.5.9/yass-ios-release-arm64-1.5.9.ipa) and [Android](https://github.com/Chilledheart/yass/releases/download/1.5.9/yass-android-release-arm64-1.5.9.apk) (ipa and apk files url). While I tested it on iOS 17 and Android 14, it worked preliminarily as a basic vpn client in the mobile device. Basically, you can test it on iOS 13 or later and Android 7 or later. Please watch out for the full [release note](https://github.com/Chilledheart/yass/releases/tag/1.5.9) before you try it.
> 
> Internally, it uses the same SSL stack code from chromium but builds in cmake. I tried to make it as close as chromium but at minimal efforts. It also supports some legacy ciphers from shadowsock but not that much. With the "http2" method, I used it as an alternative to SagerNet in android and Shadowrocket in ios. It is a minimal clone of those GUI clients so it doesn't contain many high level features. Stay tuned and report to me if you want other people to have more support.
> 
> Below is the screenshot of the android client. The screenshot is in Chinese, but don't worry. Of course, it use English as basic language.
> 
> <img alt="snapshot-android" width="484" src="https://private-user-images.githubusercontent.com/54673341/292949555-50d0f9ea-cdcf-409d-830e-9ac846489e14.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyODcyMDIsIm5iZiI6MTcwNDI4NjkwMiwicGF0aCI6Ii81NDY3MzM0MS8yOTI5NDk1NTUtNTBkMGY5ZWEtY2RjZi00MDlkLTgzMGUtOWFjODQ2NDg5ZTE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDEzMDE0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU4NmE3NTAxZTFkZWRiNDFiOGVmN2YxZGVhNWExMTg2ZjQxNjExZjZlZTRhNjBkMzY0YTc1OGYwYTA2NTUxZGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lqeounT4I4DTujDQ5RAHbVhk9kSAUh7qz0HatMI-HHM">
> The app is quite ugly and basic. It just works as sample as naiveproxy cli command line. For the full family members (it supports windows, linux and macOS), you can look at [screenshot](https://github.com/Chilledheart/yass/wiki/Supported-Operating-System#screenshot-on-android-vpn-support).

For your reference, you can use [tun2proxy] and pass the naiveproxy's proxy url on the top of it. It will handle both of TCP and UDP packets and the whole network get processed by naiveproxy. If you want to develop something similar to my app on Android based on naiveproxy, you can try it.

[tun2proxy]: https://github.com/blechschmidt/tun2proxy

**Comment by U201413497** [comment]
> hi all, i managed to make some naiveproxy-compatible GUI clients both on [iOS](https://github.com/Chilledheart/yass/releases/download/1.5.9/yass-ios-release-arm64-1.5.9.ipa) and [Android](https://github.com/Chilledheart/yass/releases/download/1.5.9/yass-android-release-arm64-1.5.9.apk) (ipa and apk files url). While I tested it on iOS 17 and Android 14, it worked preliminarily as a basic vpn client in the mobile device. Basically, you can test it on iOS 13 or later and Android 7 or later. Please watch out for the full [release note](https://github.com/Chilledheart/yass/releases/tag/1.5.9) before you try it.
> 
> Internally, it uses the same SSL stack code from chromium but builds in cmake. I tried to make it as close as chromium but at minimal efforts. It also supports some legacy ciphers from shadowsock but not that much. With the "http2" method, I used it as an alternative to SagerNet in android and Shadowrocket in ios. It is a minimal clone of those GUI clients so it doesn't contain many high level features. Stay tuned and report to me if you want other people to have more support.
> 
> Below is the screenshot of the android client. The screenshot is in Chinese, but don't worry. Of course, it use English as basic language.
> 
> <img alt="snapshot-android" width="484" src="https://private-user-images.githubusercontent.com/54673341/292949555-50d0f9ea-cdcf-409d-830e-9ac846489e14.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MzM1NjYsIm5iZiI6MTcwNDQzMzI2NiwicGF0aCI6Ii81NDY3MzM0MS8yOTI5NDk1NTUtNTBkMGY5ZWEtY2RjZi00MDlkLTgzMGUtOWFjODQ2NDg5ZTE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA1VDA1NDEwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThmMzM4M2IwNTVlODEyNDAyYWRhNTM0ZDcyZGUyZWNkNmQ4NmZiOWMyZDliY2M4MzNhNTllZTNmMGFiMTYwYTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.X7sYEs9jIjEEr_C6x7OuO047uPYeH3QR6wYqphNLQcU">
> The app is quite ugly and basic. It just works as sample as naiveproxy cli command line. For the full family members (it supports windows, linux and macOS), you can look at [screenshot](https://github.com/Chilledheart/yass/wiki/Supported-Operating-System#screenshot-on-android-vpn-support).

I think it's very neat and I like it. It would be perfect if it could save multiple servers and have offloading rules built in such as geoip and geosite. In addition, how to install it on ios? I saw the ipa file, but I don't know how to use it.

**Comment by Chilledheart** [comment]
> I think it's very neat and I like it. It would be perfect if it could save multiple servers and have offloading rules built in such as geoip and geosite. In addition, how to install it on ios? I saw the ipa file, but I don't know how to use it.

Thanks.

It might be supported to use multiple offload rules. I need modify the current framework to support multiple upstreams at first.

For the iOS, the most common way is getting app installed from AppStore and TestFlight, but not from ipa files. In the latest progress, I faced some errors when applying testflight external test url and it blocks me from any further attempt. I might ask Apple Support and publish the testflight external test url once it is done.

**Comment by Chilledheart** [comment]
> > I think it's very neat and I like it. It would be perfect if it could save multiple servers and have offloading rules built in such as geoip and geosite. In addition, how to install it on ios? I saw the ipa file, but I don't know how to use it.
> 
> Thanks.
> 
> It might be supported to use multiple offload rules. I need modify the current framework to support multiple upstreams at first.
> 
> For the iOS, the most common way is getting app installed from AppStore and TestFlight, but not from ipa files. In the latest progress, I faced some errors when applying testflight external test url and it blocks me from any further attempt. I might ask Apple Support and publish the testflight external test url once it is done.

And you might give iTunes or Xcode a try to install IPA files directly. It seems possible to install IPA files according to this https://medium.com/@itskaranzzz/how-to-install-ipa-file-directly-to-iphones-tabs-without-using-laptop-desktop-e645c36125d2

**Comment by klzgrad** [comment]
I have moved the clients recognized by me to the main README.md for better visibility, and thus this issue is no longer needed.

The criteria for inclusion and mentions in my README.md are these:

* Open-source
* Related to NaiveProxy code or the protocol
* Actively maintained: has serious releases within a year, can ask for exceptions if you believe it's feature matured and needs no maintenance.

You can also create new issues to announce new projects or request to add it to README.md. I will leave the issues open for those projects that are not serious enough to meet the above criteria for inclusion.
